File: writersblock-back/src/main/java/com/muratoni/writersblock/service/StoryCharService.java
package com.muratoni.writersblock.service;

import java.util.List;

import com.muratoni.writersblock.model.StoryChar;

public interface StoryCharService {

    StoryChar createStoryChar(StoryChar place);

    List<StoryChar> getAllStoryChars();

    List<StoryChar> getStoryCharsByName(String name);

    List<StoryChar> getStoryCharsExcept(List<Long> ids);

    List<StoryChar> getStoryCharsByNameExcept(String name, List<Long> tIds);

    StoryChar updateStoryChar(Long id, StoryChar storyChar);

    boolean deleteStoryChar(Long id);
    
}

File: writersblock-front/src/services/StoryCharService.js
import axios from "axios";

const CHARACTER_API_BASE_URL = "http://localhost:8080/api/v1/storychar"

class StoryCharService{
    saveStoryChar(storyChar){
        return axios.post(CHARACTER_API_BASE_URL, storyChar);
    }

    getStoryChars(){
        return axios.get(CHARACTER_API_BASE_URL + "s");
    }

    getStoryCharsByName(name){
        return axios.get(CHARACTER_API_BASE_URL + "s/" + name);
    }

    getStoryCharsExcept(ids){
        const newIds = ids.join(",");
        return axios.get(CHARACTER_API_BASE_URL + "s/except/" + newIds);
    }

    getStoryCharsByNameExcept(name, ids){
        const newIds = ids.join(",");
        return axios.get(CHARACTER_API_BASE_URL + "s/name_except", {params: {
            name: name,
            ids: newIds
        }});
    }

    updateStoryChar(id, storyChar){
        return axios.put(CHARACTER_API_BASE_URL + "/" + id, storyChar);
    }

    deleteStoryChar(id){
        return axios.delete(CHARACTER_API_BASE_URL + "/" + id);
    }
}

export default new StoryCharService();